src/Grpc.AspNetCore.Server/Internal/CallHandlers/ServerCallHandlerBase.cs

```diff
@@ -134,6 +137,29 @@ protected void DisableMinRequestBodyDataRateAndMaxRequestBodySize(HttpContext ht
         }
     }
 
+#if NET8_0_OR_GREATER
+    protected void DisableRequestTimeout(HttpContext httpContext)
+    {
+        // Disable global request timeout on streaming methods.
+        var requestTimeoutFeature = httpContext.Features.Get<IHttpRequestTimeoutFeature>();
+        if (requestTimeoutFeature is not null)
+        {
+            // Don't disable if the endpoint has explicit timeout metadata.
+            var endpoint = httpContext.GetEndpoint();
+            if (endpoint is not null)
+            {
+                if (endpoint.Metadata.GetMetadata<RequestTimeoutAttribute>() is not null ||
```

Comment:
Good idea!

I don't think we'd want to include this in SignalR though since it's designed for long running request.